pkg/stanza/fileconsumer/internal/trie/trie_test.go

```diff
@@ -198,3 +198,359 @@ func runTest(t *testing.T, trie *Trie, tc *testCase) {
+										},
+									},
+								},
+							},
+							"Done:ABCxyz": opTree{ // Finish reading ABCxyz and remove from trie
+								ops: []testOp{
+									del("ABCxyz", true, "just confirmed ABCxyz exists"),
+									has("ABCDEF", true, "ABCDEF should not have been deleted"),
+								},
+								continuations: map[string]opTree{
+									"Done:ABCDEF": opTree{ // Finish reading ABCDEF and remove from trie
+										ops: []testOp{
+											del("ABCDEF", true, "just confirmed ABCDEF exists"),
+										},
+									},
+									"Found:ABCDEFxyz": opTree{ // Next poll finds ABCDEFxyz
+										ops: []testOp{
+											has("ABCDEFxyz", true, "recognize ABCDEF w/ xyz appended"),
+											put("ABCDEFxyz"), // TODO how do we know we need to call this?
+										},
+										continuations: map[string]opTree{
+											"Done:ABCDEFxyz": opTree{ // Finish reading ABCDEFxyz and remove from trie
+												ops: []testOp{
+													del("ABCDEFxyz", true, "just confirmed ABCDEFxyz exists"),
+												},
+											},
+										},
+									},
+								},
+							},
+							"Found:ABCxyz,ABCDEFxyz": opTree{ // Next poll finds ABCxyz and ABCDEFxyz
+								ops: []testOp{
+									has("ABCDEFxyz", true, "recognize ABCDEF w/ xyz appended"),
+									put("ABCDEFxyz"), // TODO how do we know we need to call this?
+									has("ABCxyz", true, "ABCxyz should not have been affected"),
+								},
+								continuations: map[string]opTree{
+									"Done:ABCxyz": opTree{ // Finish reading ABCxyz and remove from trie
+										ops: []testOp{
+											del("ABCxyz", true, "just confirmed ABCxyz exists"),
+											has("ABCDEFxyz", true, "ABCDEFxyz should not have been deleted"),
+										},
+										continuations: map[string]opTree{
+											"Done:ABCDEFxyz": opTree{ // Finish reading ABCDEFxyz and remove from trie
+												ops: []testOp{
+													del("ABCDEFxyz", true, "just confirmed ABCDEFxyz exists"),
+												},
+											},
+										},
+									},
+									"Done:ABCDEFxyz": opTree{ // Finish reading ABCDEFxyz and remove from trie
+										ops: []testOp{
+											del("ABCDEFxyz", true, "just confirmed ABCDEFxyz exists"),
+											has("ABCxyz", true, "ABCxyz should not have been deleted"),
+										},
+										continuations: map[string]opTree{
+											"Done:ABCxyz": opTree{ // Finish reading ABCxyz and remove from trie
+												ops: []testOp{
+													del("ABCxyz", true, "just confirmed ABCxyz exists"),
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+					"Found:ABC,ABCDEFxyz": opTree{ // Next poll finds ABC and ABCDEFxyz
+						ops: []testOp{
+							has("ABCDEFxyz", true, "recognize ABCDEF w/ xyz appended"),
+							put("ABCDEFxyz"), // TODO how do we know we need to call this?
+							has("ABCDEF", false, "ABCDEF should have been pushed down to ABCDEFxyz"),
```

Comment:
It's true that `ABCDEF` was pushed down, but we still have `ABC` in trie. 
If we match `ABCDEF`, it will recognize `ABC \w DEF appended`